src/test/scala/chiselTests/experimental/ProgrammaticPortsSpec.scala

```diff
@@ -0,0 +1,41 @@
+// See LICENSE for license details.
+
+package chiselTests
+package experimental
+
+import chisel3._
+
+// NOTE This is currently an experimental API and subject to change
+class PrivatePort extends NamedModuleTester {
+  private val port = expectName(IO(Input(UInt(8.W))), "foo")
+  port.suggestName("foo")
+}
+
+class PortAdder(module: NamedModuleTester, name: String) {
```

Comment:
Can you add a comment here indicating the class compositional pattern (I initially interpreted this as a roundabout way to have shared functions)